### PR TITLE
Minimum version handling

### PR DESCRIPTION
**Pull request by rottabonus**
### What has changed?
Minimum version handling for mobile client
- If not minimum version, then direct user to store
- Store-url is a build configuration variable (different for ios and android app)

Testing has been done by Felix and Eppu with builds from AppCenter ✅

### Why was the change made?
Currently we are forced to be backwards compatible. We need to implement this so we can be somewhat sure that users are using up to date applications as time passes.

### Caveats?
NA

### Related Trello issue
- [Mobile Trello ticket](https://trello.com/c/kRLX9vDX/938-version-updating-deprecation-strategy)
- [API Trello ticket](https://trello.com/c/tnqkDIvh/959-version-updating-deprecation-api)

### Checklist
- [x] I have updated relevant documentation in READMES

### CODE REVIEW

**Review by pheis — CHANGES_REQUESTED** [review]
Katotaaks puhelulla lapi? Liian monta rivia :D

File: src/state/reducers/minimumVersion.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  if (clientVersion.major > requiredVersion.major) return true; // eslint-disable-next-line padding-line-between-statements
+  if (clientVersion.major < requiredVersion.major) return false;
+
+  // if major version equal, cmpare minor
+  if (clientVersion.minor > requiredVersion.minor) return true; // eslint-disable-next-line padding-line-between-statements
+  if (clientVersion.minor < requiredVersion.minor) return false;
+
+  // compare patch
+  return clientVersion.patch >= requiredVersion.patch;
+};
```
Comment (pheis) [inline src/state/reducers/minimumVersion.ts:58]: Add return type to fn;

This is tooo complex this should be just
```
return clientVersion.major >= requiredVersion.major
&& clientVersion.minor >= requiredVersion.minor && clientVersionPatch >= requiredVersionPatch
```

Comment (rottabonus) [reply src/state/reducers/minimumVersion.ts:58]: it wont work like this.
What if the version is 3.0.1 and minimum required is 2.19.0?

Comment (rottabonus) [reply src/state/reducers/minimumVersion.ts:58]: Imo this function is not complex at all. 
1. check major
2. if equal, check minor
3. if equal check patch

File: src/api/minimumVersion.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export type AppClient = {
+  client: string;
+  major: number;
+  minor: number;
+  patch: number;
+};
+
+export const toAppClient = (value: Client): AppClient => {
+  const splitted = value.version.split('.').map(Number);
```
Comment (pheis) [inline src/api/minimumVersion.ts:31]: destructure?

File: src/lib/validators.ts [hunk 3]
```diff
@@ -52,6 +52,10 @@ export const isValidDescription = (value: string): boolean => {
   return /^(?!\s*$).+/.test(value);
 };
 
+const isValidVersion = (value: string): boolean => {
+  return /^\d{1,2}\.\d{1,2}\.\d{1,2}$/.test(value);
```
Comment (pheis) [inline src/lib/validators.ts:56]: could be just \d+ or something don't quote me on regex...

**Review by pheis — DISMISSED** [review]

**Review by idolgov — APPROVED** [review]

**Pull request merged**